Panic when diffing 2 tables with same name and different primary key sets

I have the repository available on our dolthub DEV enviroment, ping me for the link.

* I created menu_items on `master` and made a bunch of changes.
* Then I checked out a new branch from master, and rolled it back to its first commit:
     * dolt checkout -b rollback
     * dolt reset --hard <first_commit_in_log>
*  And added `menu_items` with a new primary key set

Diffing the branches, you'll get the panic on table diff for menu_items:

```
$ dolt diff master km/menu_items_schema
diff --dolt a/menu_items b/menu_items
--- a/menu_items @ gv62bil4vluq4o62p2nl5vgah1b0oj77
+++ b/menu_items @ c1ahdmftsdum70jdtkhdmc3e236gc9bp
  CREATE TABLE menu_items (
    `name` VARCHAR(255) NOT NULL
-   `restaurant_id` INT NOT NULL
-   `calories` INT DEFAULT NULL
-   `fat(g)` INT DEFAULT NULL
-   `carbohydrates` INT DEFAULT NULL
-   `protein` INT DEFAULT NULL
-   `sodium(mg)` INT DEFAULT NULL
+   `restaurant_name` VARCHAR(100) NOT NULL
+   `identifier` VARCHAR(255) NOT NULL
+   `calories` INT DEFAULT NULL
+   `fat(g)` FLOAT DEFAULT NULL
+   `carbohydrates` INT DEFAULT NULL
+   `protein` INT DEFAULT NULL
+   `sodium(mg)` INT DEFAULT NULL
panic: primary key sets must be the same

goroutine 1 [running]:
github.com/dolthub/dolt/go/cmd/dolt/commands.tabularSchemaDiff(0x2a8e258, 0xc0001a4db0, 0xc0007cacee, 0xa, 0xc0007c83b0, 0xa, 0xc0000544e0, 0xc000054840, 0xc00049e900, 0x1, ...)
        /Users/katiemcculloch/Documents/dolt/go/cmd/dolt/commands/diff.go:529 +0x2267
github.com/dolthub/dolt/go/cmd/dolt/commands.diffSchemas(0x2a8e258, 0xc0001a4db0, 0xc000617b60, 0xc000617bc0, 0xc0007cacee, 0xa, 0xc0007c83b0, 0xa, 0xc0000544e0, 0xc000054840, ...)
        /Users/katiemcculloch/Documents/dolt/go/cmd/dolt/commands/diff.go:473 +0x36d
github.com/dolthub/dolt/go/cmd/dolt/commands.diffUserTables(0x2a8e258, 0xc0001a4db0, 0xc000617b60, 0xc000617bc0, 0xc000113db0, 0xc000617bc0, 0xc000113db0)
        /Users/katiemcculloch/Documents/dolt/go/cmd/dolt/commands/diff.go:438 +0x4a5
github.com/dolthub/dolt/go/cmd/dolt/commands.DiffCmd.Exec(0x2a8e258, 0xc0001a4db0, 0xc0004188d0, 0x9, 0xc00003c0a0, 0x2, 0x2, 0xc0001f8420, 0x9)
        /Users/katiemcculloch/Documents/dolt/go/cmd/dolt/commands/diff.go:163 +0x265
github.com/dolthub/dolt/go/cmd/dolt/cli.SubCommandHandler.Exec(0x2794e77, 0x4, 0x27ab17d, 0x11, 0xc000292280, 0x25, 0x25, 0x0, 0x2a8e258, 0xc0001a4db0, ...)
        /Users/katiemcculloch/Documents/dolt/go/cmd/dolt/cli/command.go:165 +0x4c2
main.runMain(0x0)
        /Users/katiemcculloch/Documents/dolt/go/cmd/dolt/dolt.go:294 +0x136c
main.main()
        /Users/katiemcculloch/Documents/dolt/go/cmd/dolt/dolt.go:129 +0x2a
```

--- a/go/cmd/dolt/commands/diff.go
+++ b/go/cmd/dolt/commands/diff.go
@@ -526,7 +526,8 @@ func tabularSchemaDiff(ctx context.Context, td diff.TableDelta, fromSchemas, toS
 	}
 
 	if !schema.ColCollsAreCompatible(fromSch.GetPKCols(), toSch.GetPKCols()) {
-		panic("primary key sets must be the same")
+		pkSetVErr := errhand.BuildDError("cannot diff tables with different primary key sets").Build()
+		return errhand.VerboseError(pkSetVErr)
 	}
 	pkStr := strings.Join(fromSch.GetPKCols().GetColumnNames(), ", ")
 	cli.Print(sqlfmt.FmtColPrimaryKey(4, pkStr))

--- a/integration-tests/bats/diff.bats
+++ b/integration-tests/bats/diff.bats
@@ -460,3 +460,32 @@ SQL
     [ $status -eq 0 ]
     [[ $output =~ "CONSTRAINT \`fk_named\` FOREIGN KEY (\`cv1\`) REFERENCES \`parent\` (\`pv1\`)" ]] || false
 }
+
+@test "diff: table with same name on different branches with different primary key sets" {
+    dolt branch another-branch
+    dolt sql <<SQL
+CREATE TABLE a (
+  id int PRIMARY KEY,
+  pv1 int,
+  pv2 int
+);
+SQL
+    dolt add -A
+    dolt commit -m "hi"
+    dolt checkout another-branch
+    dolt sql <<SQL
+CREATE TABLE a (
+  id int primary key,
+  cv1 int primary key,
+  cv2 int
+);
+SQL
+    dolt add -A
+    dolt commit -m "hello"
+    run dolt diff master another-branch
+    echo $output
+    ! [[ "$output" =~ "panic" ]] || false
+    [[ "$output" =~ "pv1" ]] || false
+    [[ "$output" =~ "cv1" ]] || false
+    [ $status -eq 0 ]
+}

